### PR TITLE
[WIP] [CR] Alter all giant animals to drop unhealthy meat

### PR DESCRIPTION
**Pull request by Davi-DeGanne**
#### Summary
```SUMMARY: Balance "Alter all giant animals to drop unhealthy meat"```

#### Purpose of change
Butchering a single giant animal can easily feed a survivor for a long item if they can preserve all the meat. Given the abundance of these creatures, caloric concerns are little more than a chore for most of game for players that know where to look (which is to say, swamps, spider basements, and ant colonies).

#### Describe the solution
This makes meat dropped by giant mutant animals still filling, but disincentivizes long-term reliance by giving the meat a very low hidden health stat. Descriptions were modified to make mutant meat sound sketchy, but not inedible. Mutant meat can be cooked, smoked, dried, or turned into jerky, but not made into any other recipes. It has a `healthy` value of -10 for each 400 calories (approximately). The different variants (cooked, smoked, etc) have enjoyability values from -3 to 0.

#### Technical explanation of health effects
If the health modifier (the thing directly affected by comestibles) is between -10 and 10, it doesn't change over time. So after your first piece of cooked mutant meat, you're at -10 and you'll stay there until you do something to increase your health. The effects at -10 health are:

- 5% slower sleep healing
- 2.5% slower first aid healing
- get sick ~20% more often

Mostly inconsequential. So what if you eat more mutant meat? Well, if you're eating less than 2000 calories a day worth of mutant meat, very little; when your health modifier gets below -10, it naturally increases by 1 per half-hour, or by about 48 points per day. Your actual health value moves towards your health modifier, so it will hover around -10 to -15, so the effects will be mostly similar to those listed above.

If you are consuming **more** than 2000 calories of mutant meat a day, then you are outpacing the increase of 48 per day, and will eventually reach -30, where the health modifier will start increasing by 2 points every half-hour, or 96 per day. The effects at -30 health are more serious, but still not a deal-breaker by any means:

- 15% slower sleep healing
- 7.5% slower first aid healing
- get sick ~35% more often

As a final side note, leukocyte breeder basically locks your health at 100 so the health penalty becomes meaningless at that point.

#### Describe alternatives you've considered
A lot. Mostly ideas from people on the Discord, and a few of my own. I'll list as many as I can think of here (in no particular order):

- Make a new vitamin representing some weird alien cells/compound that causes disease effects at high doses.
- Remove some/all existing vitamins to make it less nutritious.
- Reduce the calories, under the logic that the meat is malformed due to rapid growth
- Reduce the enjoyability drastically, under the same logic (awful texture)
- Reduce hidden health stat even more
- Expand the mutant meat to non-giant mutants (experimental mutant, evolved mutant, gozu)
- Expand the mutant meat to nether creatures that currently drop normal meat (Gracken, flaming eye, yugg, etc)
- Give spider eggs same treatment

#### Additional context
As you can see, there are a lot of possible directions to go with this, so I was hoping get some feedback, to see what we can reach a consensus on.

To reiterate: The goal is to make mutant meat unappealing, even dangerous, in the long-term, while keeping it as a viable means of staving off hunger in the short-term.

### CODE REVIEW

**Comment by TheMurderUnicorn** [comment]
=\ I don't know how to feel about this.  I recognize the problem with the availability, but I think -15 health is obscene.  Health values in general are out of whack but a chunk of fried fat only gives -1 health.  Hell, eating straight up sewage is at -20 and your meat is right there with it.  

I don't see why a giant animal would suddenly become an absolute poison for us to eat.  

Again, I recognize the problem with meat availability, but I don't like this solution.  

Not to mention my general problem with risk/reward and accomplishment.  If I kill a giant naked mole rat, I earned the right to eat well for days.  Just my opinions.

**Comment by shadowsapex** [comment]
Imo it would be just fine if you drop the health hit down to -5 or even -3 or something. As now, it's probably not a viable option aside from 0.01% of the time. Even super early game carnivores would rather eat fat.

**Comment by Night-Pryanik** [comment]
First of all we need to understand why animals became gigantic. The answer is simple - Blob made them mutate and drastically increase in size. So gigantic animals should have their meat infested by the Blob, just as zombies have. Thus eating meat of the giant animals almost equals eating zombie flesh, with all its negative consequences. I suggest going farther and making meat of all giant animals not unhealthy, but tainted instead.

**Comment by TheMurderUnicorn** [comment]
Right, the blob makes them mutate, but I thought that the blob infected everything, no matter what.  What makes them being big different from the regular animals?  Wouldn't a regular horse produce tainted meat as well?

**Comment by DaviBones** [comment]
> First of all we need to understand why animals became gigantic. The answer is simple - Blob made them mutate and drastically increase in size. So gigantic animals should have their meat infested by the Blob, just as zombies have. Thus eating meat of the giant animals almost equals eating zombie flesh, with all its negative consequences. I suggest going farther and making meat of all giant animals not unhealthy, but tainted instead.

I actually agree with you, that was my first idea, but I-am-erk and Kevin vetoed it when we discussed it in #lore in the dev discord channel:

> I-am-erk 10/10/19 at 3:32 PM
> However tainted meat kind of represents rotten flesh powered by blob, while giant bugs are more like the phenomutants that we get by giving people mutagen. They're blob-powered just like a mutated player, but they are likely not literally made of blob.

So we arrived at this solution instead.

> I recognize the problem with the availability, but I think -15 health is obscene.
It's only -15 for the raw version, it's -10 for cooked. Compare that to a corn dog, which is -12 uncooked and -4 cooked (for the same 400 calories).

> As now, it's probably not a viable option aside from 0.01% of the time.
If you mean "not a viable option for a primary food source," well, that's kinda the whole point. If you mean "not a viable option for a temporary food source," then I suspect you're overvaluing the health stat, and/or misunderstanding how it works.

**Comment by shadowsapex** [comment]
There is no point. There's no difficulty in finding food even if this source was taken out. There's 0.01% of situations that would make this the choice you pick for food as opposed to everything else. Only a very early game carnivore would have problems and even so they'd just eat chunks of fat. Mutant meat gives like a -45 health debuff for the equivalent calories of fat.

**Comment by foulman** [comment]
If this is the case, shouldn't mutant NPCs all be unhealthy to eat as well? This doesn't take the mutant NPC mod into account, or NPCs being mutated later, dying, being eaten, correct?

**Comment by kevingranade** [comment]
The rationale for this (which won't appear in the description or anything, maybe in some lab notes) is that in order to make these animals giant, the blob had to mutate them in bizarre ways, rendering them alien and therefore unhealthy to eat.

**Comment by DaviBones** [comment]
>The rationale for this (which won't appear in the description or anything, maybe in some lab notes) is that in order to make these animals giant, the blob had to mutate them in bizarre ways, rendering them alien and therefore unhealthy to eat.

To expand on this, here's a more detailed justification that was offered by I-am-erk in #lore on the dev discord:


> ... they were unpredictably mutated by a hostile alien entity using extradimensional tricks bordering on magic. They're probably partially compatible with our biology now, and full of weird mutant cells things that can attack and interact with you in unpredictable ways. They're packed with unknown exotic chemicals, some from alien worlds, potentially even using forms of matter we're not familiar with.
>
>Or they could just be mushy and gross from growing quickly, creating a loose packed spongiform meat that is both disgusting in texture and not very nutritious.
>
>Probably all of the above.

File: data/json/monsters/reptile_amphibian.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "armor_cut": 8,
     "vision_day": 3,
     "vision_night": 35,
-    "harvest": "animal_large_noskin",
+    "harvest": "mutant_animal_large_noskin",
```
Comment (kevingranade) [inline data/json/monsters/reptile_amphibian.json]: Gator isn't a mutant, this isn't even a particularly large specimen.
```suggestion
    "harvest": "animal_large_noskin",
```

Comment (DaviBones) [reply data/json/monsters/reptile_amphibian.json]: Whoops, good catch, I think I misread the description.

**Comment by Maleclypse** [comment]
The corollary to this that I'm not seeing anyone mention is that all Post Threshold Mutations that can eat meat would be adapted to negate any meat unhealthiness in mutated meat.  The Threshold mutations specifically call out that they are adaptations to the world as it exists and the direction it is moving.  Plus anyone who is threshold mutated is basically the same amount of blob only with more control than a giant animal.

**Comment by DaviBones** [comment]
> 
> 
> The corollary to this that I'm not seeing anyone mention is that all Post Threshold Mutations that can eat meat would be adapted to negate any meat unhealthiness in mutated meat. The Threshold mutations specifically call out that they are adaptations to the world as it exists and the direction it is moving. Plus anyone who is threshold mutated is basically the same amount of blob only with more control than a giant animal.

Again, the reason mutant meat is unhealthy is **not** because it has too much blob in it (even eating pure blob globs will not hurt your health). It is because the meat is malformed from growing too quickly and has alien compounds (besides blob) in it that are not meant to be inside human.

That said, I'm not against the idea of a post-threshold mutation that disables the health penalty from mutant meat, but it's sort of out of scope for this PR, and I've no interest in implementing it myself in a different PR either.

**Comment by DaviBones** [comment]
Going in a different direction with this, will open a new PR soon.

**Pull request closed without merging**